runtime/src/bank.rs

```diff
@@ -3257,6 +3281,49 @@ impl Bank {
         self.set_compute_budget(compute_budget);
     }
 
+    fn reconfigure_token2_native_mint(self: &mut Bank) {
+        let reconfigure_token2_native_mint = match self.operating_mode() {
+            OperatingMode::Development => true,
+            OperatingMode::Preview => self.epoch() == 95,
```

Comment:
MUST, well, this gate must also check for slot index like `self.get_epoch_and_slot_index(bank.slot()) == (95, 0)`. That's because `apply_feature_activations` could be called in the middle of epoch in the case of snapshot restoration. So, we must specify the exact slot to store the updated mint account. On the other hand, other existing codes called from `apply_feature_activations` does their job nevertheless in the middle of epoch for proper iniitalization unlike account updates. The difference is that account updates are persisted by definition, other `struct` fields are not.

Sorry, maybe my comment didn't stressed enough for the requirement of being idempotent: https://github.com/solana-labs/solana/pull/11966/files#diff-a7549f152920d85fb44e6a784b8e5e1fR3227

Reply:
Hhm, actually, I now think the current code is ensured to be idempotent correctly..